wallet/wallet.h

```diff
@@ -114,4 +123,31 @@ bool wallet_can_spend(struct wallet *w, const u8 *script,
  */
 s64 wallet_get_newindex(struct lightningd *ld);
 
+/**
+ * wallet_shachain_init -- wallet wrapper around shachain_init
+ */
+bool wallet_shachain_init(struct wallet *wallet, struct wallet_shachain *chain);
+
+/**
+ * wallet_shachain_add_hash -- wallet wrapper around shachain_add_hash
+ */
+bool wallet_shachain_add_hash(struct wallet *wallet,
+			      struct wallet_shachain *chain,
+			      shachain_index_t index,
+			      const struct sha256 *hash);
+
+/* Note: not wrapping shachain_get_hash since it doesn't touch the DB */
+#define wallet_shachain_get_hash(w,chain,index,hash) \
+	shachain_get_hash(&chain->chain, index, hash);
```

Comment:
static inline ... is probably better; in particular this one is bit messed up because "chain->chain" will become "x->x" if the chain arg is "x" (it's probably always called chain, so it "works").  And you'd need (chain)->chain anyway...